email_totals/app.py

```diff
@@ -346,51 +353,86 @@ def build_summary(target_period, compare_period, team_sage):
     would be included in the 'resources' subkey and the 'accounts' subkey can
     be removed.
 
+    The 'unowned' key will map to a dictionary keyed on account IDs, and
+    each account ID will have a 'total' and 'change' subkey representing the
+    total cost of unowned resources in the account and the percent change from
+    the previous month, respectively.
+
     Example output
     ```
-    user1@example.com:
-        resources:
-            111122223333:
-                total: 10.0
-                change: 1.2
-            222233334444:
-                total: 0.1
-                change: 0.0
-        missing_other_tag:
-            111122223333:
-              - i-0abcdefg
-        accounts:
-            333344445555:
-                total: 20.0
-                change: -2.1
-    user2@example.com:
-        ...
+    account_names:
+        111122223333: account-one
+        222233334444: account-two
+        333344445555: account-three
+    per_user_summary:
+        user1@example.com:
+            resources:
+                111122223333:
+                    total: 10.0
+                    change: 1.2
+                222233334444:
+                    total: 0.1
+                    change: 0.0
+            missing_other_tag:
+                111122223333:
+                  - i-0abcdefg
+            accounts:
+                333344445555:
+                    total: 20.0
+                    change: -2.1
+        user2@example.com:
+            ...
+    unowned:
+        111122223333:
+            total: 1.2
+            change: 0.0
     ```
     """
 
     data = {}
     min_value = float(os.environ['MINIMUM'])
 
-    # Generate 'resources' subkeys and merge them in
+    # Generate 'resources' subkeys
     resources = get_resource_totals(target_period, compare_period, min_value)
+    LOG.debug(f"Resource data: {resources}")
+
+    # Unowned resource costs will be associated with an empty string owner,
+    # use pop() to remove the unowned data from the dictionary.
+    # While IT-2369 is blocked the data will also include account totals for
+    # accounts tagged with an owner, remove them as they are discovered.
+    unowned = {}
+    if '' in resources:
```

Comment:
this is checking for a key equal to the empty string. the keys in this dictionary are the owner email addresses returned by cost explorer, and cost explorer will report an owner equal to the empty string for uncategorized costs. then the lambda includes the entry for the empty string just like the entries for valid addresses.

here's an example `resources` dict:
```
email1@example.com:
  resources:
    111122223333:
      total: 101.0
      change: 0.01
email2@example.com:
  resources:
    111122223333:
      total: 99.0
      change -0.01
'':
  resources:
    111122223333:
      total: 100.0
      change: 0.0
``` 
and that shows entries for two users and an entry for uncategorized costs in account.

previously the empty-string entries were being filtered out by the `valid_recipients` function.

Reply:
thanks for the explanation, it was not apparent to me that `get_resource_totals` method would return a dict with an empty key.  i think you should really describe add this explanation to the docstring of that method.  

Also the use of `resources` as a variable here is a bit confusing since there is also a static resources key in the returned dictionary.